### PR TITLE
Cmake support

### PR DESCRIPTION
**Pull request by Ezbob**
Instead having an specific Linux Makefile, this CMakeLists.txt can generate both Visual Studio build scripts and an Makefile for Linux.
Some documentation is put in the README file.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -53,3 +53,4 @@ dkms.conf
 
 .vscode
 game
+/build*/
```
Comment (Tiggles) [inline .gitignore]: What is the `*`for? Is it for the CMake output based on platform?

File: README.md [hunk 2]
```diff
@@ -9,3 +9,24 @@
 - Quit pressing the "Escape" key!
 - Experience dynamic allocation of new wrappers when placing too many houses!
 - Experience some of the houses not being correctly rendered if you move too far away from your starting point!
+
+### Building with cmake
+
+This project has an CMakeLists.txt setup for generating a plethora of build scripts for build systems
```
Comment (Tiggles) [inline README.md]: has _a_ CMakeList.txt setup

**Review by Tiggles — CHANGES_REQUESTED** [review]
Need to make sure why `/build*/` is used. If a source folder called `buildings`, this could be an issue.

File: .gitignore (hunk 1 above)
Comment (Ezbob) [reply .gitignore]: It's there for if you want to generate build script files f.x. clang and visual studio on the same machine. Then you can generate something like `build.clang` for the clang compiler, and `build.vstudio` for the visual studio compiler.
It also just matches `build` if you don't bother with multiple toolchains.
All of these paths are local to the root .gitignore file (that's what the prefix / does)

File: README.md (hunk 2 above)
Comment (Ezbob) [reply README.md]: Right, one of my bad habits.

Comment (Ezbob) [reply README.md]: Should be fixed now

File: .gitignore (hunk 1 above)
Comment (Tiggles) [reply .gitignore]: Wouldn't `/build.*/` be a better fit?

Comment (Ezbob) [reply .gitignore]: Then it wont fit just `build/` ... Anyway, I can make it into `/build/` and then just use sub folders within that, to build for different compilers. Is that an OK solution?

Comment (Tiggles) [reply .gitignore]: So `/build/clang` for instance? That's perfectly fine.

Comment (Ezbob) [reply .gitignore]: Yes. Alright, I'll fix it

**Review by Tiggles — APPROVED** [review]

**Comment by Tiggles** [comment]
Seems to work on MacOS as well

**Comment by Ezbob** [comment]
Nice! 
Well, CMake should be intelligent enough to find out what OS it is run on and use the "default" toolchain for that OS.
The real problem is getting the dependencies, but you can use a dependency manager such as Vcpkg or Conan to fetch those if you want (I did that to get the project running on Windows 10).

**Pull request merged**